apis/vmware/v1beta1/providerserviceaccount_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ProviderServiceAccountSpec defines the desired state of ProviderServiceAccount.
+type ProviderServiceAccountSpec struct {
+	// Ref specifies the reference to the VSphereCluster for which the ProviderServiceAccount needs to be realized.
+	Ref *corev1.ObjectReference `json:"ref"`
+
+	// Rules specifies the privileges that need to be granted to the service account.
+	Rules []rbacv1.PolicyRule `json:"rules"`
+
+	// TargetNamespace is the namespace in the target cluster where the secret containing the generated service account
+	// token needs to be created.
+	TargetNamespace string `json:"targetNamespace"`
+
+	// TargetSecretName is the name of the secret in the target cluster that contains the generated service account
+	// token.
+	TargetSecretName string `json:"targetSecretName"`
+}
+
+// ProviderServiceAccountStatus defines the observed state of ProviderServiceAccount.
+type ProviderServiceAccountStatus struct {
+	Ready    bool   `json:"ready,omitempty"`
+	ErrorMsg string `json:"errorMsg,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=providerserviceaccounts,scope=Namespaced
```

Comment:
we do have a `categories=cluster-api` for other types. just makes it easier to look up the resources I think

Reply:
Makes sense, was missing from the original PR and hence that label did not get included. Will update the PR with the label for these types.